node/tests/utils.rs

```diff
@@ -171,66 +171,75 @@ impl MASQNode {
+            log_output_processor,
             limit_ms,
         );
     }
 
-    //gives back all possible captures by given requirements;
-    //you can specify how many times the regex needs to be looked for;
-    //also allows to define multiple capturing groups and fetch them all at once (the inner vector)
+    //fetches all captures by given requirements;
+    //you can specify how many times you require to apply the regex and get separate captures (outer vector);
+    //also allows to define more than just one capture group and fetch them all at once (inner vector)
     pub fn capture_pieces_of_log_at_directory(
         pattern: &str,
         logfile_dir: &Path,
-        required_number_of_captures: usize,
+        required_repetition: usize,
         limit_ms: Option<u64>,
     ) -> Vec<Vec<String>> {
         let logfile_path = Self::path_to_logfile(logfile_dir);
 
-        let do_with_log_output = |log_output: &String, regex: &Regex| -> Option<Vec<Vec<String>>> {
-            let captures = regex
-                .captures_iter(&log_output[..])
-                .collect::<Vec<Captures>>();
-            if captures.len() < required_number_of_captures {
-                return None;
-            }
-            let structured_captures = (0..captures.len())
-                .flat_map(|idx| {
-                    captures.get(idx).map(|capture| {
-                        (0..capture.len())
-                            .flat_map(|idx| {
-                                capture.get(idx).map(|particular_group_match| {
-                                    particular_group_match.as_str().to_string()
-                                })
-                            })
-                            .collect::<Vec<String>>()
-                    })
-                })
-                .collect::<Vec<Vec<String>>>();
-            Some(structured_captures)
+        let log_output_processor = |log_output: &str, regex: &Regex| {
+            Self::collect_captures_with_repetition(log_output, regex, required_repetition)
         };
 
         Self::wait_for_log_at_directory(
             pattern,
             logfile_path.as_ref(),
-            &do_with_log_output,
+            log_output_processor,
             limit_ms,
         )
         .unwrap()
     }
 
-    fn wait_for_log_at_directory<T>(
+    fn collect_captures_with_repetition(
+        log_output: &str,
+        regex: &Regex,
+        required_repetition: usize,
+    ) -> Option<Vec<Vec<String>>> {
+        let captures = regex.captures_iter(log_output).collect::<Vec<Captures>>();
+        if captures.len() < required_repetition {
+            return None;
+        }
+        let structured_captures = (0..captures.len())
+            .flat_map(|idx| {
+                captures.get(idx).map(|capture| {
+                    (0..capture.len())
+                        .flat_map(|idx| {
+                            capture.get(idx).map(|particular_group_match| {
```

Comment:
Play the same trick here. If Regex tells you there are N captures and then doesn't want to give you all of them, that should be a panic.